File: src/SmartTrack.Scripts/Scripts/Watermark.cs
using System;
using jQueryApi;

namespace SmartTrack.Scripts
{
    public static class Watermark
    {
        public static void spanBlur(jQueryObject span)
        {
            if (span.Siblings("input").GetValue().Trim().Length == 0)
                span.Show().CSS("z-index", "100");
        }

        public static void spanFocus(jQueryObject span)
        {
            span.Hide();
            span.Siblings("input").Focus();
        }

        public static void inputBlur(jQueryObject input)
        {
            if (input.GetValue().Trim().Length == 0)
                input.Siblings("span").Show().CSS("z-index", "100");
        }

        public static void inputFocus(jQueryObject input)
        {
            input.Siblings("span").Hide();
        }
    }
}

File: src/SmartTrack/Scripts/watermark.js
(function() {
  var Watermark;
  Watermark = (function() {
    function Watermark() {}
    Watermark.prototype.blurSpan = function(span) {
      if (span.siblings('input').val().trim().length === 0) {
        return span.show().css('z-index', 100);
      }
    };
    Watermark.prototype.focusSpan = function(span) {
      span.hide();
      return span.siblings('input').focus();
    };
    Watermark.prototype.blurInput = function(input) {
      if (input.val().trim().length === 0) {
        return input.siblings('span').show().css('z-index', 100);
      }
    };
    Watermark.prototype.focusInput = function(input) {
      return input.siblings('span').hide();
    };
    return Watermark;
  })();
  window.Watermark = new Watermark();
  $(function() {
    return $('.watermark').blur();
  });
}).call(this);
